src/rnnlm/rnnlm-compute-state.cc

```diff
@@ -0,0 +1,131 @@
+// src/rnnlm/rnnlm-compute-state.cc
+
+// Copyright      2017  Johns Hopkins University (author: Daniel Povey)
+//                2017  Yiming Wang
+//                2017  Hainan Xu
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rnnlm/rnnlm-compute-state.h"
+#include "nnet3/nnet-utils.h"
+#include "nnet3/nnet-compile-looped.h"
+
+namespace kaldi {
+namespace rnnlm {
+
+RnnlmComputeStateInfo::RnnlmComputeStateInfo(
+    const RnnlmComputeStateComputationOptions &opts,
+    const kaldi::nnet3::Nnet &rnnlm,
+    const CuMatrix<BaseFloat> &word_embedding_mat):
+    opts(opts), rnnlm(rnnlm), word_embedding_mat(word_embedding_mat) {
+  KALDI_ASSERT(IsSimpleNnet(rnnlm));
+  int32 left_context, right_context;
+  ComputeSimpleNnetContext(rnnlm, &left_context, &right_context);
+  KALDI_ASSERT(0 == left_context);
```

Comment:
I think it would be better to check the conditions with a single if, and use KALDI_ERR.  That way the check will be active even in non-debug mode (this matters becaue it could happen due to user error, not just code error).  The same for the embedding-dim check.  Give a nice informative error message there -- it's a likely scripting problem.